code/clue.js

```diff
@@ -1,59 +1,408 @@
+
+  weaponCard.style.background = mystery.weapon.color;
+  weaponName.innerHTML = `${mystery.weapon.name}`;
+  weaponWeight.innerHTML = `${mystery.weapon.weight}`;
+  weaponImage.src = `${mystery.weapon.image}`;
+};
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
+
+  const roomCard = document.getElementById("roomCard");
+  const roomName = document.getElementById("roomName");
+  const roomImage = document.getElementById("roomImage");
+
+  roomName.style.display = "block";
+  roomImage.style.display = "block";
+  document.getElementById("loadingAnimation3").style.display = "none";
+
+  roomCard.style.background = mystery.room.color;
+  roomName.innerHTML = `${mystery.room.name}`;
+  roomImage.src = `${mystery.room.image}`;
+};
+
+const startAnimation1 = () => {
+  const killerImage = document.getElementById("killerImage");
+  const favouriteWeapon = document.getElementById("favouriteWeapon");
+  const killerAge = document.getElementById("killerAge");
+  const killerName = document.getElementById("killerName");
+
+  killerImage.style.display = "none";
+  favouriteWeapon.style.display = "none";
+  killerAge.style.display = "none";
+  killerName.style.display = "none";
+  document.getElementById("loadingAnimation1").style.display = "block";
+
+  setTimeout(pickKiller, 1500)
+};
+
+const startAnimation2 = () => {
+  const weaponImage = document.getElementById("weaponImage");
+  const weaponName = document.getElementById("weaponName");
+  const weaponWeight = document.getElementById("weaponWeight");
+
+  weaponName.style.display = "none";
+  weaponWeight.style.display = "none";
+  weaponImage.style.display = "none";
+  document.getElementById("loadingAnimation2").style.display = "block";
+
+  setTimeout(pickWeapon, 1500)
+};
+
+const startAnimation3 = () => {
+  const roomName = document.getElementById("roomName");
+  const roomImage = document.getElementById("roomImage");
+
+  roomName.style.display = "none";
+  roomImage.style.display = "none";
+  document.getElementById("loadingAnimation3").style.display = "block";
+
+  setTimeout(pickRoom, 1500)
+};
+
+// CALL FUNCTIONS WHEN CLICK ON CARDS
+
+killerCard.onclick = startAnimation1;
+weaponCard.onclick = startAnimation2;
+roomCard.onclick = startAnimation3;
+
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+// FINAL FUNCTION
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const revealMystery = () => {
+  const answerMystery = document.getElementById("mysteryAnswer")
+  if (mystery.killer === undefined || mystery.weapon === undefined || mystery.room === undefined) {
```

Comment:
Nice adding this condition to make sure all cards are clicked before revealing the mystery! 😎

Reply:
Agreed! Well done!